### PR TITLE
Added List manager, steps and features.

### PR DESCRIPTION
**Pull request by jorgeposterla**
```
Feature: Trello API Lists # tests/api/features/lists.feature:1
  As a user
  I want "Lists" API endpoints
  In order to manipulate lists
  Scenario: GET all lists in a board                      # tests/api/features/lists.feature:6
    Given I created a board                               # tests/api/features/steps/lists_steps.py:8
    And I created a list on the board with name "list 22" # tests/api/features/steps/lists_steps.py:13
    When I send a "GET" request to "boards/{board}/lists" # tests/api/features/steps/common_steps.py:11
    Then I receive a list with at least "1" "list"        # tests/api/features/steps/common_steps.py:18
    And the status code is "200"                          # tests/api/features/steps/common_steps.py:29

  Scenario: GET a list by id                              # tests/api/features/lists.feature:13
    Given I created a board                               # tests/api/features/steps/lists_steps.py:8
    And I created a list on the board with name "list 22" # tests/api/features/steps/lists_steps.py:13
    When I send a "GET" request to "/lists/{list}"        # tests/api/features/steps/common_steps.py:11
    Then I receive a response with the "list" id          # tests/api/features/steps/common_steps.py:23
    And the status code is "200"                          # tests/api/features/steps/common_steps.py:29

  Scenario: POST a new list to a board                 # tests/api/features/lists.feature:20
    Given I created a board                            # tests/api/features/steps/lists_steps.py:8
    When I send a "POST" request to "/lists"           # tests/api/features/steps/common_steps.py:11
      | Key     | Value      |
      | name    | list 1     |
      | idBoard | {id}:board |
    Then I receive a response with the "list" schema   # tests/api/features/steps/common_steps.py:34
    And the status code is "200"                       # tests/api/features/steps/common_steps.py:29
    When I send a "GET" request to "/lists/{response}" # tests/api/features/steps/common_steps.py:11
    Then I check the data is                           # tests/api/features/steps/lists_steps.py:19
      | Key     | Value      |
      | name    | list 1     |
      | idBoard | {id}:board |

  Scenario: Update a list name                              # tests/api/features/lists.feature:34
    Given I created a list on the board with name "list 22" # tests/api/features/steps/lists_steps.py:13
    When I send a "PUT" request to "/lists/{list}"          # tests/api/features/steps/common_steps.py:11
      | Key  | Value  |
      | name | list33 |
    Then I check the data is                                # tests/api/features/steps/lists_steps.py:19
      | Key  | Value  |
      | name | list33 |
    And the status code is "200"                            # tests/api/features/steps/common_steps.py:29

  Scenario: Archive a list                                  # tests/api/features/lists.feature:44
    Given I created a list on the board with name "list 22" # tests/api/features/steps/lists_steps.py:13
    When I send a "PUT" request to "/lists/{list}"          # tests/api/features/steps/common_steps.py:11
      | Key    | Value |
      | closed | true  |
    Then I check the data is                                # tests/api/features/steps/lists_steps.py:19
      | Key    | Value |
      | closed | true  |
    And the status code is "200"                            # tests/api/features/steps/common_steps.py:29
```

### CODE REVIEW

File: tests/api/features/steps/lists_steps.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+from main.utils.behave_helpers import fill_payload
+
+manager = ListsManager()
+
+
+@given('I created a board')
+def step_impl(context):
+    pass
+
+
```
Comment (mefemefe) [inline tests/api/features/steps/lists_steps.py:12]: There already a step in master's board_steps.py with the same purpose ('A board is created')
```suggestion

```

File: tests/api/features/steps/lists_steps.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+@step('I check the data is')
+def step_impl(context):
+    expected_json = fill_payload(context, payload={})
+    for key, value in expected_json.items():
+        actual_value = str(context.response[key])
+        if key == 'closed':
+            actual_value = actual_value.lower()
+        assert str(value) == actual_value, \
+            f'expected value for {key} is {value}, but actual is {actual_value}'
```
Comment (mefemefe) [inline tests/api/features/steps/lists_steps.py:27]: This step already exists with the same name in master's common_steps.py, with a different implementation.
```suggestion
```
If the idea is to change the same step, it shouldn't be part of lists_steps.

File: tests/api/features/steps/lists_steps.py (hunk 1 above)
Comment (jorgeposterla) [reply tests/api/features/steps/lists_steps.py]: Fixed in new commit.

File: tests/api/features/steps/lists_steps.py (hunk 2 above)
Comment (jorgeposterla) [reply tests/api/features/steps/lists_steps.py]: Fixed in new commit.

**Review by mefemefe — APPROVED** [review]

**Review by trab2 — APPROVED** [review]

File: tests/api/features/lists.feature [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        And the status code is "200"
+
+    Scenario: GET a list by id
+        Given A board is created
+        And I created a list on the board with name "list 22"
+        When I send a "GET" request to "/lists/{list}"
+        Then I receive a response with the "list" "id"
+        And the status code is "200"
+
+    Scenario: POST a new list to a board
```
Comment (mefemefe) [inline tests/api/features/lists.feature:20]:
```suggestion
    @smoke
    Scenario: POST a new list to a board
```

**Pull request merged**